contracts/tgrade-valset/src/contract.rs

```diff
@@ -540,7 +546,27 @@ fn end_block(deps: DepsMut, env: Env) -> Result<Response, ContractError> {
     let old_validators = VALIDATORS.load(deps.storage)?;
     VALIDATORS.save(deps.storage, &validators)?;
     // determine the diff to send back to tendermint
-    let (diff, update_members) = calculate_diff(validators, old_validators);
+    let (diff, add, remove) = calculate_diff(validators, old_validators);
+    let update_members = RewardsDistribution::UpdateMembers {
```

Comment:
Hmm... Why not just return this as it was before (same function signature for `calculate_diff `) and iterate over 
`for op in update_members.add` for example? It should avoid a clone even.

Reply:
Good idea.

Reply:
We'd end up in this situation:
A function called `calculate_diff` actually calculates the rewards distribution msg for use with a specific contract's interface, and then we use that message to get back the diff for another use.

I'd rather make `calculate_diff` do what it advertises in the first place. Feels less convoluted.

As for the cloning, I think we'd still end up cloning those strings (`Addr::unchecked` does that), except they'd be cloned implicitly rather than explicitly. I could be missing something.

Reply:
Makes sense, i.e. moving the rewards distribution out of `calculate_diff` is a good thing:tm:. Let's merge as it is.